tests/test_other.py

```diff
@@ -2346,18 +2346,24 @@ def test_embind_closure_no_dynamic_execution(self):
     create_file('post.js', '''
       Module['onRuntimeInitialized'] = function() {
         out(Module['foo'](10));
+        out(Module['bar']());
       };
     ''')
     create_file('main.cpp', r'''
+      #include <string>
       #include <emscripten/bind.h>
       int foo(int x) { return x; }
+      std::string bar() {
+        return emscripten::val(123).call<std::string>("toString");
```

Comment:
Is it not enough to just `return "123"` here?   i.e. is the fix for the JS->wasm directly or the wasm->JS direction or both?

Reply:
The bug does not happen without it. I'm not familiar enough with embind details to understand why though.

Reply:
Fair enough.. I does the bug happen if you just call `emscripten::val(123).call<std::string>("toString");` within the existing bound `foo`?   i.e. do you need to declare and bind a string-returning C function to tickly this bug it is it enough to just call `"toString"` on something?

lgtm either way.